### PR TITLE
Fix runpath_search_paths for inherit search_path targets

### LINKED ISSUE #8337: UI Tests no longer run after Firebase/Core pod install
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

When I installed the Firebase Core pod and tried running my test target it runs my unit tests correctly but fails to launch my UI tests.

I raised this issue with the Firebase team [here](https://github.com/firebase/firebase-ios-sdk/issues/2147) and they recommended its a CocoaPods issue.

I found one workaround solutions which is to add `$(FRAMEWORK_SEARCH_PATHS)` to the UI Test target's Runpath search paths. Which is also suggested [here](https://stackoverflow.com/questions/34990175/test-target-x-encountered-an-error-early-unexpected-exit-operation-never-finis).

## What did you do?

Try run my test target after a I installed Firebase/Core pod.

## What did you expect to happen?

My Unit and UI tests to run.

## What happened instead?

It failed to launch my UI tests giving the following error:

BrokenFirebaseCoreUITests-Runner.app (80714) encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted. (Underlying error: The test runner failed to load the test bundle. Executable cannot be loaded for some other reason, such as a problem with a library it depends on or a code signature/entitlements mismatch.))

### Stack

```
   CocoaPods : 1.5.3
        Ruby : ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin18]
    RubyGems : 2.7.6
        Host : Mac OS X 10.14.1 (18B75)
       Xcode : 10.1 (10B61)
         Git : git version 2.19.1
Ruby lib dir : /Users/apolk/.rbenv/versions/2.5.1/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 67b896ef7d187eccb6f5088812330091490ad3d4
```

### Installation Source

```
Executable Path: /Users/apolk/.rbenv/versions/2.5.1/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'BrokenFirebaseCore' do
  # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
  use_frameworks!

  # Pods for BrokenFirebaseCore
  pod 'Firebase/Core', '5.13.0'

  target 'BrokenFirebaseCoreTests' do
    inherit! :search_paths
    # Pods for testing
  end

  target 'BrokenFirebaseCoreUITests' do
    inherit! :search_paths
    # Pods for testing
  end

end
```

## Project that demonstrates the issue

A test project that replicates the issue can be found [here](https://github.com/firebase/firebase-ios-sdk/files/2648116/BrokenFirebaseCore.zip). Simply run the test target (CMD+U) to see the error.

### PR DESCRIPTION
**Pull request by paulb777**
Fix #8337

### CODE REVIEW

**Comment by domdop** [comment]
Ich bin derzeit nicht im Büro. Ich kehre zurück am  02.01.2019.
Ich werde Ihre Nachricht nach meiner Rückkehr beantworten.

Die Rückmeldung bezieht sich auf ein Mail mit folgendem Thema:
[CocoaPods/CocoaPods] Fix runpath_search_paths for inherit search_path
targets (#8381)

____________________________________________________________________________________________
Gesendet (c) GRZ/RACON Linz 2018 Agent 'Abwesenheit'

----------------------------------------
Raiffeisen Software GmbH
Firmenbuchnummer: 86804d, Landesgericht Linz
DVR: 0572934, UID: ATU 36961105

Der Austausch von Nachrichten mit dem oben angeführten Absender via E-Mail
dient ausschließlich Informationszwecken. Rechtsgeschäftliche Erklärungen
dürfen über dieses Medium nicht ausgetauscht werden.
Correspondence with above mentioned sender via e-mail is only for
information purposes. This medium is not to be used for the exchange of
legally-binding communications.
----------------------------------------

File: CHANGELOG.md [hunk 1]
```diff
@@ -53,6 +53,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
+* Fix runpath_search_paths for inherit !search_path targets
```
Comment (dnkoutso) [inline CHANGELOG.md]: nit: "for `inherit! :search_paths` can also use backticks for formatting.

**Comment by dnkoutso** [comment]
@paulb777 two questions:

a) Can you explain more why the framework search paths need to be part of the runpath search paths? I thought all frameworks are copied to the same default destination which then are loaded at runtime.

b) Do you want this for 1.6.x or 1.7.x?

**Comment by paulb777** [comment]
@dnkoutso:

a) It seems to be the case that the frameworks in the host application's app bundle are not sufficient for UITests. See the example at #8337 and the discussion at https://stackoverflow.com/questions/34990175/test-target-x-encountered-an-error-early-unexpected-exit-operation-never-finis

b) This one could go into 1.6.x if it doesn't contribute to a longer delay. Let me know and I can do the merge.  I don't feel too strongly since this seems to be a long-standing issue with workarounds.

File: CHANGELOG.md [hunk 2]
```diff
@@ -53,7 +53,7 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
-* Fix runpath_search_paths for inherit !search_path targets  
+* Fix runpath_search_paths for `inherit !search_path` targets
```
Comment (dnkoutso) [inline CHANGELOG.md]: nit of the nits you can ignore me but I think its `inherit!` not `inherit !`

**Comment by dnkoutso** [comment]
will mark for 1.7.0 lets keep the 1.6.0 close to being done.

File: CHANGELOG.md (hunk 2 above)
Comment (paulb777) [reply CHANGELOG.md]: Thanks! Fixed and squashed.

**Pull request merged**